### PR TITLE
ref: Remove CultureUiInfo if values are even with CultureInfo

### LINKED ISSUE #669: Improve reporting "Current Culture"
image

- [ ] Rename key to have a space (i.e: "Current Culture" instead of "CurrentCulture"
- [ ] If both values are the same, don't report "Current UI Culture"

### PR DESCRIPTION
**Pull request by lucas-zimerman**
Closes #669

### CODE REVIEW

**Comment by lucas-zimerman** [comment]
TaskUnobservedTaskExceptionIntegration.cs | 76.92% <0.00%> (-7.70%)
-- | --
There's no connection between the changes and UnobTaskExpintegration O_o

File: src/Sentry/Internal/MainSentryEventProcessor.cs [hunk 1]
```diff
@@ -80,18 +83,23 @@ public SentryEvent Process(SentryEvent @event)
...
+            {
+                cultureInfoMapped = new Dictionary<string, string>();
             }
 
-            const string cultureInfoKey = "CurrentCulture";
-            if (!@event.Contexts.ContainsKey(cultureInfoKey)
-                && CultureInfoToDictionary(CultureInfo.CurrentCulture) is { } currentCultureMap)
+            if (!@event.Contexts.ContainsKey(CurrentUiCultureKey)
+                && CultureInfoToDictionary(CultureInfo.CurrentUICulture) is { } currentUiCultureMap
+                && currentUiCultureMap.Any(p => cultureInfoMapped.Contains(p) is false))
```
Comment (bruno-garcia) [inline src/Sentry/Internal/MainSentryEventProcessor.cs:100]: Just realized there's finally a `TryAdd`:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.tryadd?view=net-5.0

So we could avoid the double lookup, but would require adding a polifyll for ns2.0 and net4x.

Also, `is false`, maybe stay with `!` instead?
```suggestion
                && !currentUiCultureMap.Any(p => cultureInfoMapped.Contains(p)))
```

**Review by bruno-garcia — COMMENTED** [review]
Thanks for another PR! Left a note thopugh

File: src/Sentry/Internal/MainSentryEventProcessor.cs [hunk 2]
```diff
@@ -80,18 +83,23 @@ public SentryEvent Process(SentryEvent @event)
...
+            if (!@event.Contexts.ContainsKey(CultureInfoKey)
+                && CultureInfoToDictionary(CultureInfo.CurrentCulture) is { } currentCultureMap)
             {
-                @event.Contexts[currentUiCultureKey] = currentUiCultureMap;
+                cultureInfoMapped = currentCultureMap;
+                @event.Contexts[CultureInfoKey] = currentCultureMap;
+            }
+            else
+            {
+                cultureInfoMapped = new Dictionary<string, string>();
```
Comment (bruno-garcia) [inline src/Sentry/Internal/MainSentryEventProcessor.cs:95]: Don't allocate here

File: src/Sentry/Internal/MainSentryEventProcessor.cs [hunk 3]
```diff
@@ -80,18 +83,23 @@ public SentryEvent Process(SentryEvent @event)
...
             }
 
-            const string cultureInfoKey = "CurrentCulture";
-            if (!@event.Contexts.ContainsKey(cultureInfoKey)
-                && CultureInfoToDictionary(CultureInfo.CurrentCulture) is { } currentCultureMap)
+            if (!@event.Contexts.ContainsKey(CurrentUiCultureKey)
+                && CultureInfoToDictionary(CultureInfo.CurrentUICulture) is { } currentUiCultureMap
+                && currentUiCultureMap.Any(p => !cultureInfoMapped.Contains(p)))
```
Comment (bruno-garcia) [inline src/Sentry/Internal/MainSentryEventProcessor.cs:100]:
```suggestion
                && cultureInfoMapped is null || currentUiCultureMap.Any(p => !cultureInfoMapped.Contains(p)))
```

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

**Review by bruno-garcia — APPROVED** [review]
Nice!

CI isn't happy, flaky test?

**Comment by lucas-zimerman** [comment]
> Nice!
> 
> CI isn't happy, flaky test?

Only on windows, it's failing to restore the Visual Studio packages.

**Pull request merged**